src/specBuilder/specUtils.ts

```diff
@@ -296,4 +296,21 @@ export const usePreviousChartData = <T>(data: T) => {
 	}, [data]);
   
 	return previousDataRef.current;
-};
\ No newline at end of file
+};
+
+export const getAnimationMarks = (dimension: string, metric: string, data?: ChartData[], previousData?: ChartData[]) => {
+	let markUpdate = {
+		scale: "yLinear",
+		signal: `datum.${metric} * timerValue`
+	};
+	if (data && previousData) {
+		const isCongruent = data.every((d) => previousData.some((pd) => d[dimension] === pd[dimension]));
```

Comment:
`isCongruent` is a nice short name, but I didn't immediately understand it.  Could we try `hasSameDimensions`?  Then it's more clear what we are branching on on the next line